### PR TITLE
Fix luck animation offset

### LINKED ISSUE #975: Morale and luck animation should be shifted lower
Currently morale and luck animation for units standing in the upper part of battlefield goes out of visible part of battlefield.
Morale 1

In the OG we can see whole animation:
Morale

### PR DESCRIPTION
**Pull request by idshibanov**
Fixes #975 . 

Making sure luck animations are fully displayed even in the top row. I missed this one last time :)

### CODE REVIEW

File: src/fheroes2/battle/battle_interface.cpp [hunk 1]
```diff
@@ -3262,18 +3265,25 @@ void Battle::Interface::RedrawActionLuck( Unit & unit )
...
 
             if ( frameId < 8 && Battle::AnimateInfrequentDelay( Game::BATTLE_MISSILE_DELAY ) ) {
                 RedrawPartialStart();
 
                 const Sprite & luckSprite = AGG::GetICN( ICN::CLOUDLUK, frameId );
-                luckSprite.Blit( pos.x + pos.w / 2 + luckSprite.x(), pos.y + pos.h + luckSprite.y() - 10, _mainSurface );
+
+                int y = pos.y + pos.h + luckSprite.y() - 10;
+                if ( y < 0 ) {
+                    y = maxHeight + luckSprite.y();
```
Comment (ihhub) [inline src/fheroes2/battle/battle_interface.cpp:3283]: Will the animation then jump at certain frames? Imagine that at first 3 frames y is more than 0 due to smaller cloud and then on forth y is negative leading to a shift of this animation center.

Comment (idshibanov) [reply src/fheroes2/battle/battle_interface.cpp]: You are right, could happen. Corrected.

File: src/fheroes2/battle/battle_interface.cpp [hunk 2]
```diff
@@ -3262,18 +3265,25 @@ void Battle::Interface::RedrawActionLuck( Unit & unit )
...
 
             if ( frameId < 8 && Battle::AnimateInfrequentDelay( Game::BATTLE_MISSILE_DELAY ) ) {
                 RedrawPartialStart();
 
                 const Sprite & luckSprite = AGG::GetICN( ICN::CLOUDLUK, frameId );
-                luckSprite.Blit( pos.x + pos.w / 2 + luckSprite.x(), pos.y + pos.h + luckSprite.y() - 10, _mainSurface );
+                int y = pos.y + pos.h - 10;
+
+                // move drawing position if it will clip outside of the battle window
+                if ( y - maxHeight < 0 )
```
Comment (ihhub) [inline src/fheroes2/battle/battle_interface.cpp:3283]: I think we need to move all this code for `y` calculation out of the loop as it doesn't depend on **luckSprite**.

**Comment by ihhub** [comment]
@idshibanov , thank you very much for fixing this issue as well!

**Pull request merged**